### PR TITLE
feat: the sentence model arrives on the first English dictation

### PR DESCRIPTION
**Pull request by znat**
The first English dictation now starts a background download of ModernBERT from
`znaat/modernbert-coreml`, about 300 MB. It is compiled once and the compiled
form is kept under `~/Library/Application Support/ParrotFlow/models/`. The pill
and the menu bar show the progress the way they already show the speech model.

Nothing reads the model yet. A later PR adds the tokenizer and the sentence
boundary probe. The fetch is not awaited, so the dictation that starts it is
never held up.

Check two things first. One, `Status.downloading` gains a `blocking` flag —
without it a dictation started mid-fetch shows the download on its pill.
Two, `HubDownload` is a new downloader; FluidAudio's `ModelHub` could not be
used.

Verify with `ParrotFlow --sentence-model`. Run it twice: the second run says
`cached` and skips both the download and the compile.

<details>
<summary>The pill claim was the one real race, and it needed a flag to fix</summary>

The task was to check that `.ready` and the label token still clear when the
download outlives the dictation that started it. They do. `pillDownloadRun` is
only set at `AppDelegate.swift:1839` when a dictation starts while a download is
already running, and the sentence fetch starts after transcription, so it never
claimed the pill of the run that began it. `transcriberLabelToken` clears on
`.ready` as before.

The break is one dictation later. Hold the hotkey again while the sentence model
is still coming down and `transcribe()` reads `transcriberStatus`, finds
`.downloading`, and puts `Downloading sentence model 62%` on that dictation's
pill instead of `Transcribing…`. That dictation is not waiting on anything. The
same status also set `permissions.model.speechModel = .preparing`, which tells
the setup window the speech model has not arrived and you cannot speak yet.

So `Status.downloading` now carries `blocking: Bool`. Three consumers read it:

| Site | With `blocking: false` |
|---|---|
| `AppDelegate.swift:1839` | no pill claim |
| `AppDelegate.swift:5195` | no `updateProgress` |
| `AppDelegate.swift:5208` | no `speechModel = .preparing` |

The menu bar label is written either way, which is the progress line this PR is
for.

The fetch also reports through `onStatusChange` without writing
`Transcriber.status`. `status` answers "can this transcribe", and during a
background fetch it can. When the fetch ends it re-sends whatever `status`
actually holds, so a blocking download running at the same time keeps its label
and its pill.

A progress report can also arrive late. The callback reaches the actor through
an unstructured task, so a percentage could land after the fetch had already put
the real status back, leaving a finished download's number in the menu bar.
`Transcriber.sentenceModelRunning` drops a report that arrives after the fetch
ended; both halves run on the actor, so they are ordered.

One case is left. While the sentence fetch runs it overwrites the menu bar label
of a concurrent vocabulary download. Both downloads have to be in flight at once
for it to happen, which is at most once per install, and only the menu bar text
is affected — the pill and the setup window are correct.

</details>

<details>
<summary>A truncated 299 MB weight file loads and answers nonsense, so it is checked</summary>

Every file goes to a temporary path, gets its length checked, and only then
moves into a staging directory inside the cache directory. Staging is inside the
cache directory on purpose: `moveItem` across volumes copies rather than
renames, and a copy is not atomic. The compiled model moves in last, so
`isCached` is never true for a half-built cache.

The length comes from `https://huggingface.co/api/models/<repo>/tree/main`, not
from `Content-Length` on the download. HuggingFace serves `tokenizer.json`
gzipped, and a gzipped response carries no length at all:

    $ curl -sIL -H 'Accept-Encoding: gzip' .../resolve/main/tokenizer.json
    HTTP/2 200
    content-encoding: gzip
    (no content-length)

The tree call gives the real byte count for all four files in one request.

A cache that is damaged after the fact cannot be recovered by catching the load
error. Deleting `model.mil` from the compiled model segfaults inside CoreML:

    EXC_BAD_ACCESS (SIGSEGV) at 0x0
    CoreML makeProgramWithMemoryLayout(...)
    CoreML -[MLE5ProgramLibrary prepareAndReturnError:]
    CoreML +[MLLoader loadModelFromAssetAtURL:configuration:error:]

So `isCached` checks that `model.mil`, `coremldata.bin` and
`weights/weight.bin` are all present before anything is loaded. With that check
the same damage re-downloads instead of crashing. Measured both ways.

A process that quits mid-download leaves a `.staging-<uuid>` directory holding
up to 299 MB. The next fetch deletes any it finds. Killed with `kill -9` ten
seconds in, the cache directory holds this and nothing else:

    drwxr-xr-x  .staging-6E61F6DD-0E0C-43F9-A325-60F13CD0BB75
    296K  total

No `ModernBERT-base-64.mlmodelc`, so `isCached` is false and nothing can load
it. The next run cleaned the staging directory up and finished in 43.4s.

Two processes share this cache: the app and `--sentence-model`. Before the fix
below, the second one's staging cleanup deleted the first one's live download,
and the first died with `A valid manifest does not exist at path:
.../.staging-C32FE695/ModernBERT-base-64.mlpackage/Manifest.json`. `fetch` now
runs under an exclusive `flock` on `models/modernbert-base-64.lock`, beside the
cache directory rather than inside it, so clearing the cache cannot delete a
lock somebody holds. It does not wait for the lock — the second process gives
up and the next English dictation asks again.

</details>

<details>
<summary>Four things were decided rather than found, and here is why</summary>

**Our own downloader.** FluidAudio's `ModelHub.download` takes a `Repo`, which
is a closed enum in `ModelNames.swift`. It cannot name a repository outside its
own list. No new package dependency was added.

**A session delegate, not `download(from:delegate:)`.** The `delegate:` argument
of the async download only receives life-cycle and authentication callbacks.
`didWriteData` never arrives, and the fetch reports 0% then 100%. Measured: with
a real session delegate the label climbs through every percent.

**The `.mlpackage` is deleted after the compile.** Keeping it would cost another
299 MB and buy nothing — a cache that will not load is re-fetched, not rebuilt.

**Application Support, not `~/.config/parrotflow`.** The config directory holds
files a person opens in an editor. `AppVariant.supportDirectory` keeps the
dev/release split every other path here has, so a dev build cannot damage the
installed app's cache.

</details>

<details>
<summary>What this run looks like, and what the checks say</summary>

Cold, with the cache removed:

    $ ParrotFlow --sentence-model
      sentence model 0%…
      sentence model 1%…
      ... every percent ...
      sentence model 100%…
    ✓ sentence model ready in 43.9s (downloaded and compiled)
      cache   ~/Library/Application Support/ParrotFlow/models/modernbert-base-64
      inputs  input_ids [1, 64]
      outputs logits [1, 64, 50368]

Warm:

    ✓ sentence model ready in 0.1s (cached)

The cache is 288 MB. `make test` passes: `swift build -c release`, swiftlint,
and all 25 check scripts. swiftlint reports no new warnings — the two in
`Transcriber.swift` and `main.swift` are on `main` already.

</details>

<details>
<summary>What this does not do</summary>

- Nothing reads the model or the tokenizer. `tokenizer.json` is fetched anyway
  so the next PR needs no downloader change.
- French dictation never triggers the fetch. The gate is
  `Pipeline.language(of:config:) == "en"` on the transcript, which is the same
  call the correction prompt uses.
- `warmUpTranscriber()` is untouched. This is first use, not launch.
- There is no way to delete the cache from inside the app.

</details>

### CODE REVIEW

File: Sources/ParrotFlow/Transcriber.swift [hunk 1]
```diff
@@ -102,7 +106,47 @@ actor Transcriber {
...
+    /// in the menu bar.
+    private func warmSentenceModel() {
+        guard sentenceModelFetch == nil else { return }
+        sentenceModelFetch = Task { [weak self] in
+            guard let self else { return }
+            var failed = false
+            do {
+                try await SentenceModel.shared.prepare { label in
+                    Task { await self.reportSentenceModel(label) }
+                }
```
[the remark this answers is not in the document]
Comment (znat) [reply Sources/ParrotFlow/Transcriber.swift:137]: Right, and fixed in 0ac8e6f. The callback hops onto the actor through an unstructured task, so a percentage could land after `finishSentenceModel` had put the real status back, and the menu bar kept it.

`Transcriber` now holds `sentenceModelRunning`. `finishSentenceModel` clears it, and `reportSentenceModel` returns early when it is false. Both run on the actor, so a report that was queued before the fetch ended and executes after it is dropped.

File: Sources/ParrotFlow/SentenceModel.swift [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    /// this whole shape exists to prevent.
+    private static func fetch(progress: (@Sendable (String) -> Void)?) async throws {
+        let files = FileManager.default
+        try files.createDirectory(at: directory, withIntermediateDirectories: true)
+        // A process that quits mid-download leaves its staging directory and
+        // whatever it had fetched, which is up to 299 MB nothing will ever read.
+        let left = try? files.contentsOfDirectory(atPath: directory.path)
+        for name in left ?? [] where name.hasPrefix(".staging-") {
+            try? files.removeItem(at: directory.appendingPathComponent(name))
+        }
```
[the remark this answers is not in the document]
Comment (znat) [reply Sources/ParrotFlow/SentenceModel.swift:165]: Right, and fixed in 0ac8e6f. Reproduced it the same way: `ParrotFlow --sentence-model` started four seconds after another one deleted the first process's staging directory, and the first died with "A valid manifest does not exist at path: .../.staging-C32FE695/ModernBERT-base-64.mlpackage/Manifest.json".

`SentenceModel.fetch` now runs under an exclusive `flock` on `models/modernbert-base-64.lock`. The lock file sits beside the cache directory, not inside it, so a fetch that clears the cache cannot delete a lock somebody holds. Closing the descriptor releases it, and the kernel closes it for a process that dies holding one.

It does not wait for the lock. The fetch is background work nothing reads yet, so the second process logs and gives up, and the next English dictation asks again. Same run after the fix:

    == second process while the first is downloading:
    ✗ another ParrotFlow process is fetching the sentence model
    == waiting for the first:
    ✓ sentence model ready in 27.4s (downloaded and compiled)

The staging cleanup is now only ever run by the lock holder, which is what makes deleting every `.staging-*` safe.
